tests/run.py

```diff
@@ -235,6 +235,29 @@ def substitute_workspace_dir(jsonfile, wsdir):
     for line in fileinput.input(jsonfile, inplace=True):
         print(line.replace("#WORKSPACE_DIR#", wsdir)),
 
+def test_with_java_options(tmpdir, lib_path, jacoco):
+    sys.stdout.write("Testing with Java Options...")
+    import tarfile
+    compatDir = tmpdir+'/compat_100_test'
+    tar = tarfile.open('inputs/compatibility.pre.1.0.0.test.tar')
+    tar.extractall(path=compatDir)
+    loader_json = compatDir+'/t0_1_2.json'
+    query_json = compatDir+'/t0_1_2_java_vcf.json'
+    substitute_workspace_dir(loader_json, compatDir)
+    substitute_workspace_dir(query_json, compatDir)
+    query_command = 'java'+jacoco+' -Dgenomicsdb.library.path='+lib_path+' -DGATK_STACKTRACE_ON_USER_EXCEPTION=true -ea TestGenomicsDB --query -l '+loader_json+' '+query_json
+    pid = subprocess.Popen(query_command, shell=True, stdout=subprocess.PIPE);
+    stdout_string = pid.communicate()[0]
+    if(pid.returncode != 0):
+        sys.stderr.write('Query with java options : '+query_command+' failed\n');
+        cleanup_and_exit(tmpdir, -1);
+    actual_md5sum = str(hashlib.md5(stdout_string).hexdigest())
+    golden_str, golden_md5sum = get_file_content_and_md5sum('golden_outputs/java_t0_1_2_vcf_at_0')
+    if (actual_md5sum != golden_md5sum):
+        sys.stderr.write('Query with java options : '+query_command+' failed. Results did not match with golden output.\n');
+        cleanup_and_exit(tmpdir, -1)
+    sys.stdout.write("Successful\n")
+
```

Comment:
Question about this test...is it supposed to be testing the stack trace option is being passed in correctly and that it gets printed out? Or something else?

Reply:
It is testing only the JNI code for passing down the stack trace option more like a unit test really. I was looking at adding this to the junits, but there seems to be no clean way to get multiple class loaders in the tests. The query itself runs cleanly, so there is not exception to track.

Reply:
Gotcha - a little nitpicky, but I wonder if it is worth it to manipulate the jsons (or the workspace) to get it to throw an exception and check that?

Reply:
Done.